Test branch

alimulondo opened the pull request:
What does this PR do?

This PR Implements the test cases of the api.

Description of Task to be completed?
View questions endpoint tested
Answer to question tested
View single question tested


.

alimulondo committed:
fx([#159859235] test cases): Add test cases
Here are test cases for post question endpoints
[Fixes #159859400]

alimulondo committed:
fx(Add test cases):Modify test file
Here test cases for viewing single question are added
[Fixes #159859386]

alimulondo committed:
ft(Add CI): Add travis file
This file is for continues intergration
[Deliver #159859614]

alimulondo committed:
Add coverall file

alimulondo committed:
ft(Heroku file): Add Procfile
This file is used by heroku servers to host my my app
[Delivers #159859674]

alimulondo committed:
Modify travis file

alimulondo committed:
Modify readme

alimulondo committed:
Modify READme

alimulondo committed:
Add badges

hadijahkyampeire reviewed (approved):
LGTM

JoyLubega commented on tests/Test_base.py:
```diff
@@ -32,8 +32,34 @@ def test_question_content(self):
+
+        result = self.client.get('/api/v1/questions/{}'.format(questionId), content_type = 'application/json')
+        
+        self.assertEqual(result.status_code, 200)    
```
Avoiding testing for status codes only. Test for data

JoyLubega reviewed (changes requested):
Make your Readme detailed, its should include project setup

alimulondo committed:
Modify git ignore

alimulondo committed:
Modify gitignore

alimulondo committed:
Change questionid to start from 1

alimulondo committed:
fx(Fix add answer): Modify views
Here use can now add more than one answer to a given question
[Fixes #159944107]

alimulondo committed:
Modify tests

alimulondo committed:
Modify readme

alimulondo commented:
I have effected the changes

alimulondo committed:
Include file structure in readme.txt

alimulondo committed:
Modify readme

alimulondo committed:
Modify readme

alimulondo committed:
Add data testing to tests

alimulondo commented:
Challenge two completed
